server/src/test/java/org/opensearch/index/shard/SearchOperationListenerTests.java

```diff
@@ -286,6 +301,21 @@ public void validateReaderContext(ReaderContext readerContext, TransportRequest
         assertEquals(2, newScrollContext.get());
         assertEquals(2, freeContext.get());
         assertEquals(2, freeScrollContext.get());
+        assertEquals(0, searchIdleWakenUp.get());
+        assertEquals(0, validateSearchContext.get());
+
+        compositeListener.onNewSearchIdleWakenUp();
+        assertEquals(2, preFetch.get());
+        assertEquals(2, preQuery.get());
+        assertEquals(2, failedFetch.get());
+        assertEquals(2, failedQuery.get());
+        assertEquals(2, onQuery.get());
+        assertEquals(2, onFetch.get());
+        assertEquals(2, newContext.get());
+        assertEquals(2, newScrollContext.get());
+        assertEquals(2, freeContext.get());
+        assertEquals(2, freeScrollContext.get());
+        assertEquals(2, searchIdleWakenUp.get());
```

Comment:
Could you please explain why count increments by 2 after a single call to `compositeListener.onNewSearchIdleWakenUp()`? I assumed it was doing a single increment

Reply:
We increment 1 for each listener, and this test contains two listeners, so we increase twice.
final List<SearchOperationListener> indexingOperationListeners = new ArrayList<>(Arrays.asList(listener, listener));
indexingOperationListeners has two listeners inside